add native species Nightsiders, Sleepers, Khakturian, Celestephyte

st-pa-fo opened the pull request:
Todos @stpa:
- [x] testing
- [x] and more testing
- [x] default protection focus seems broken, the volp-ugl & nightsiders default to industry instead
- [x] more descriptive text -> enough
- [x] improve upon visuals -> solved by omitting big pictures and
- [x] additional species (plural) from the master list to be added here as well due to necessary crossreferencing and due to tediousness of keeping them all in seperate branches
- optional afterthought: add the effect to remove the star of the system where the Nightsiders are placed or change star type to none or whatever the correct expression of that idea will turn out to be
- [x] rework sleeper- and nightsider-icons
- [x] fix building reference errors
- [x] squash and rebase on master
- [x] rinse and repeat ... sorry, i could not keep from doing more native species stuff

see also https://www.freeorion.org/forum/viewtopic.php?p=109222#p109222 for the nightsiders,
https://www.freeorion.org/forum/viewtopic.php?f=7&t=12241 for the sleepers and https://www.freeorion.org/forum/viewtopic.php?f=7&t=12242 for the khakturian

<img width="755" alt="species #1 nightsiders" src="https://user-images.githubusercontent.com/37298400/149943711-671529f1-7541-4832-9277-8aacf47c2cf3.png">
<img width="731" alt="species #2 sleepers" src="https://user-images.githubusercontent.com/37298400/149943724-4dfe120b-607a-4cce-a6f9-7c6066657ba4.png">
<img width="763" alt="species #3 khakturian" src="https://user-images.githubusercontent.com/37298400/149943740-11e2ee2a-76d0-4d99-80fb-422a8fc7b389.png">
<img width="508" alt="species #4 celestephyte" src="https://user-images.githubusercontent.com/37298400/150042912-335006f5-66a5-45de-8f69-7b7d99eecd7e.png">

UPD. Make todos make more solid
edit: thx, i changed the todos around a bit and renamed the pr and added links to species-threads
edit again: updated screenshots

st-pa-fo committed:
- adding new native species #1 Nightsiders, #2 Sleepers and #3 Khakturian
- for the nightsiders see also https://www.freeorion.org/forum/viewtopic.php?p=109222#p109222
- for the sleepers see also https://www.freeorion.org/forum/viewtopic.php?f=7&t=12241
- for the khakturian see also https://www.freeorion.org/forum/viewtopic.php?f=7&t=12242
- reworked their icons so they are similar to but distinct from the preexisting species icons
- adding likes and dislikes with motivational comments to species definitions, partly translated into player readable pedia page texts
- grooming typos crossing the t's and dotting the i's
- moving the effect 'Light Sensitive' from banforo species definition to common population macros because the nightsiders share that trait
- added logging when preferred native focus not available during universe creation
- changed condition for availability of FOCUS_PROTECTION so that some native species with that same preferred focus can now do that while not yet conquored

st-pa-fo committed:
- neverending story - w.i.p. for more native species
- removing superfluos patch. oops.
- more population and environment macros in view of upcoming species
- during universe creation, allow new native species to be placed who have no good habitat but only adequate or poor
- adding species #4 the Celestephyte, to be continued ...

st-pa-fo committed:
- slightly improved formatting of pedia text for "likes" and "dislikes" and "liked by" and "disliked by"
--> but still not quite perfect: the comma must be internationalized and cannot stand for all languages. i've seen there is a japanese translation underway, the correct punctuation for that would be different than most european languages.
- fixed python formatting error: empty line is now empty
- Celestephyte descriptive pedia text (which i would later like to translate at least to german myself..)

st-pa-fo committed:
- adding tag PEDIA_PC_FUEL to laenfa species definition to be consistent with sly
- fluffing out sly description some more to account for stealthiness and possible other gaseous beings
- also give some more reason why they might dislike the gasgiant-generators
- the Thenian are not only well defended, but now have good planetary stealth too and can not colonize anyway
- new helper method LinkStringIfPossible used for hyperlinking likes and dislikes in pedia pages

st-pa-fo committed:
- streamlining the pedia text generating code a little by introducing versatile helper methods to replace some massively redundant code and prettify the player visible output at the same time, admittedly at the expense of a tiny little bit of extra computations every time the player clicks on a pedia link but that is negligible and not really noticeable

st-pa-fo committed:
- adding FieldTypes to repertoire of LinkStringIfPossible
- adding placeholder-icon for new species #5 the Thenian - and this is the last species in this pr
- the Thenian suffer from a totally natural periodic population dieoff with sitrep-message which im testing before the push
- slight reformatting of the homeworlds-part of species pedia pages to look more like the occupied planets part and to not have to scroll for miles when there are a lot of homeworlds

st-pa-fo committed:
- ending every automatically created pedia page with an extra empty line, just in case the previously last bit of text was ment to be a clickable pedia link, which would not have been clickable if the very last thing on the page. this should not hinder the player's reading experience in any way but improve it slightly, wherever pedia pages ended with pedia links before. one example in this pr is the void rift fieldtype page which lists as species that dislike the Celestephyte which is now clickable with this fix here.

st-pa-fo committed:
- fixing population dieoff effect for species Thenian and the according sitrep message
- improved sitrep-message 'new dyson forest growing' with pedia link to dyson forest (for the noobs)
- noted some todo's related to pedia with possible future improvements
- fixed some fixme's - they work best when followed by a colon, this will highlight them similar to todo's in modern ide's
- make list of homeworlds commaseparated in the pedia-homeworlds-page

agrrr3 reviewed (commented):
quite some comments, need to sleep now

st-pa-fo committed:
- adding prefixes EXTREMELY_BAD and VERY_BAD for POPULATION species-tags for use by the AI
- adding prefix VERY_BAD for RESEARCH and INDUSTRY species-tags for use by the AI - although they were already previously present in species definition of exobots

st-pa-fo committed:
- incorporating feedback about the khakturian: they do not dislike the worldtree enough to revolt anymore, but neither do they like it nor enjoy its benefits

st-pa-fo committed:
- adding system special of derelict small troop ships crewed by khakturian, to be randomly stumbled upon players just like the other derelicts

st-pa-fo committed:
- generic fieldtype pedia pages now list map-clickable known occurances of each field type

st-pa-fo committed:
- removing reference to disabled content BLD_CLONING_CENTER from species likes/dislikes
- while at it, putting species likes/dislikes each on their single line of code to facilitate commenting and merging, and apply alphabetic sorting order using "sort | uniq"

st-pa-fo committed:
- quick fix: reducing probability for natives on roids and GGs by factor five
- until there are either more candidate native species for those environments or more detailled universe setup options

st-pa-fo committed:
- rebased on master
